readme/Readme.scalatex

```diff
@@ -463,31 +559,147 @@
+                    This is due to the fact that many binary formats depends on constant-sized blocks
+                    and the easiest and fastest method to parse them is to write @hl.scala{.rep(exactly=...)}.
+                @li
+                    @b
+                        The same extensive usage of @hl.scala{.flatMap}.@br
+                    @hl.scala{flatMap} allows to retrieve data from one parser and pass it to the next.
+                    The most popular and primitive example is dynamic-sized array with length written at the beginning.
+                    Similar algorithm is used in the @hl.scala{bmp} when header information is passed to the @hl.scala{bmpRow}
+                    that returns new parser for row in this particular bmp file.
+
+        @sect{ClassParser}
+            @p
+                @div(id := "clssdiv")
+                @script(raw("""demo.DemoMain().clss(document.getElementById("clssdiv"))"""))
+                @i
+                    Given form accepts .class bytecode files for java language and prints names of all fields and methods with their descriptors.
+                    If you haven't any .class files around, you can download
+                    some examples (
+                    @a("Book.java", href:="https://github.com/lihaoyi/fastparse/raw/master/byteparse/shared/src/test/resources/Book.java"),
+                    @a("Book.class", href:="https://github.com/lihaoyi/fastparse/raw/master/byteparse/shared/src/test/resources/Book.class"),
+                    @a("Book2.java", href:="https://github.com/lihaoyi/fastparse/raw/master/byteparse/shared/src/test/resources/Book2.java"),
+                    @a("Book2.class", href:="https://github.com/lihaoyi/fastparse/raw/master/byteparse/shared/src/test/resources/Book2.class"),
+                    @a("CodeTest.java", href:="https://github.com/lihaoyi/fastparse/raw/master/byteparse/shared/src/test/resources/CodeTest.java"),
+                    @a("CodeTest.class", href:="https://github.com/lihaoyi/fastparse/raw/master/byteparse/shared/src/test/resources/CodeTest.class"))
+
+            @p
+                @a("Another byte parser", href:="https://github.com/lihaoyi/fastparse/tree/master/byteparse/shared/src/main/scala/byteparse/classparse")
+                that is present in the fastparse is @hl.scala{ClassParser}.
+                It's quite a complex parser that process .class files from java bytecode.
+                It's able to retrieve almost full information about given class including all methods, fields, subclasses, code
+                and pack it into the convenient AST.
+                On the similarity with other big parsers it has good set of unit-tests
+                with tests which compile real projects from github (joda-time, junit4, jenkins and etc.)
+                and check each built file.
+
     @sect{API Highlights}
 
-        @sect{Parser[T]}
+        @sect{Parser[T, ElemType, Repr]}
```

Comment:
No need to change this for now; the `Parser` type from `fastparse.all` is still `fastparse.all.Parser[T]`. We can talk about the more general `Parser[T, ElemType, Repl]` under the `Internal Structure` section